{"variants":[{"paths":["\/documentation\/openaikit\/openaierror\/apierror(_:)"],"traits":[{"interfaceLanguage":"swift"}]}],"kind":"symbol","sections":[],"metadata":{"fragments":[{"text":"case","kind":"keyword"},{"text":" ","kind":"text"},{"text":"apiError","kind":"identifier"},{"text":"(","kind":"text"},{"text":"APIError","kind":"typeIdentifier","preciseIdentifier":"s:9OpenAIKit8APIErrorV"},{"text":")","kind":"text"}],"symbolKind":"case","role":"symbol","roleHeading":"Case","externalID":"s:9OpenAIKit0A7AIErrorO8apiErroryAcA8APIErrorVcACmF","modules":[{"name":"OpenAIKit"}],"title":"OpenAIError.apiError(_:)"},"identifier":{"url":"doc:\/\/OpenAIKit\/documentation\/OpenAIKit\/OpenAIError\/apiError(_:)","interfaceLanguage":"swift"},"schemaVersion":{"patch":0,"minor":3,"major":0},"hierarchy":{"paths":[["doc:\/\/OpenAIKit\/documentation\/OpenAIKit","doc:\/\/OpenAIKit\/documentation\/OpenAIKit\/OpenAIError"]]},"primaryContentSections":[{"kind":"declarations","declarations":[{"platforms":["Linux"],"tokens":[{"text":"case","kind":"keyword"},{"text":" ","kind":"text"},{"text":"apiError","kind":"identifier"},{"text":"(","kind":"text"},{"text":"APIError","preciseIdentifier":"s:9OpenAIKit8APIErrorV","kind":"typeIdentifier","identifier":"doc:\/\/OpenAIKit\/documentation\/OpenAIKit\/APIError"},{"text":")","kind":"text"}],"languages":["swift"]}]},{"kind":"parameters","parameters":[{"content":[{"inlineContent":[{"type":"text","text":"The "},{"identifier":"doc:\/\/OpenAIKit\/documentation\/OpenAIKit\/APIError","type":"reference","isActive":true},{"type":"text","text":" containing error details"}],"type":"paragraph"}],"name":"error"}]},{"kind":"content","content":[{"anchor":"discussion","type":"heading","text":"Discussion","level":2},{"type":"paragraph","inlineContent":[{"text":"This case wraps the structured error response from OpenAIâ€™s API,","type":"text"},{"text":" ","type":"text"},{"text":"providing detailed information about what went wrong.","type":"text"}]},{"anchor":"Example","type":"heading","text":"Example","level":2},{"type":"codeListing","syntax":"swift","code":["do {","    let completion = try await openAI.completions.create(","        model: \"gpt-3.5-turbo\",","        messages: messages,","        maxTokens: 5000 \/\/ Too high","    )","} catch OpenAIError.apiError(let apiError) {","    print(\"Error type: \\(apiError.error.type ?? \"unknown\")\")","    print(\"Error message: \\(apiError.error.message)\")","    if let param = apiError.error.param {","        print(\"Problem parameter: \\(param)\")","    }","}"]},{"style":"note","type":"aside","name":"See Also","content":[{"type":"paragraph","inlineContent":[{"type":"text","text":""},{"identifier":"doc:\/\/OpenAIKit\/documentation\/OpenAIKit\/APIError","type":"reference","isActive":true},{"type":"text","text":", "},{"identifier":"doc:\/\/OpenAIKit\/documentation\/OpenAIKit\/APIErrorDetail","type":"reference","isActive":true}]}]}]}],"abstract":[{"type":"text","text":"An error response was received from the OpenAI API."}],"references":{"doc://OpenAIKit/documentation/OpenAIKit/OpenAIError":{"kind":"symbol","type":"topic","role":"symbol","fragments":[{"kind":"keyword","text":"enum"},{"kind":"text","text":" "},{"kind":"identifier","text":"OpenAIError"}],"url":"\/documentation\/openaikit\/openaierror","abstract":[{"text":"The primary error type for all OpenAI API interactions.","type":"text"}],"navigatorTitle":[{"kind":"identifier","text":"OpenAIError"}],"title":"OpenAIError","identifier":"doc:\/\/OpenAIKit\/documentation\/OpenAIKit\/OpenAIError"},"doc://OpenAIKit/documentation/OpenAIKit/OpenAIError/apiError(_:)":{"fragments":[{"text":"case","kind":"keyword"},{"text":" ","kind":"text"},{"text":"apiError","kind":"identifier"},{"text":"(","kind":"text"},{"text":"APIError","preciseIdentifier":"s:9OpenAIKit8APIErrorV","kind":"typeIdentifier"},{"text":")","kind":"text"}],"title":"OpenAIError.apiError(_:)","type":"topic","url":"\/documentation\/openaikit\/openaierror\/apierror(_:)","role":"symbol","identifier":"doc:\/\/OpenAIKit\/documentation\/OpenAIKit\/OpenAIError\/apiError(_:)","abstract":[{"text":"An error response was received from the OpenAI API.","type":"text"}],"kind":"symbol"},"doc://OpenAIKit/documentation/OpenAIKit":{"type":"topic","kind":"symbol","role":"collection","url":"\/documentation\/openaikit","abstract":[{"text":"A powerful, type-safe Swift SDK for the OpenAI API with support for all major endpoints and platforms.","type":"text"}],"title":"OpenAIKit","identifier":"doc:\/\/OpenAIKit\/documentation\/OpenAIKit"},"doc://OpenAIKit/documentation/OpenAIKit/APIError":{"url":"\/documentation\/openaikit\/apierror","abstract":[{"type":"text","text":"A structured error response from the OpenAI API."}],"navigatorTitle":[{"kind":"identifier","text":"APIError"}],"type":"topic","title":"APIError","fragments":[{"kind":"keyword","text":"struct"},{"kind":"text","text":" "},{"kind":"identifier","text":"APIError"}],"kind":"symbol","role":"symbol","identifier":"doc:\/\/OpenAIKit\/documentation\/OpenAIKit\/APIError"},"doc://OpenAIKit/documentation/OpenAIKit/APIErrorDetail":{"url":"\/documentation\/openaikit\/apierrordetail","abstract":[{"text":"Detailed error information from the OpenAI API.","type":"text"}],"navigatorTitle":[{"kind":"identifier","text":"APIErrorDetail"}],"type":"topic","title":"APIErrorDetail","fragments":[{"kind":"keyword","text":"struct"},{"kind":"text","text":" "},{"kind":"identifier","text":"APIErrorDetail"}],"kind":"symbol","role":"symbol","identifier":"doc:\/\/OpenAIKit\/documentation\/OpenAIKit\/APIErrorDetail"}}}